### PR TITLE
Create gauge to observe grafana config size

### PR DESCRIPTION
**Pull request by titolins**
Fixes https://github.com/grafana/alerting-squad/issues/900

### CODE REVIEW

File: pkg/services/ngalert/metrics/remote_alertmanager.go [hunk 1]
```diff
@@ -80,5 +81,12 @@ func NewRemoteAlertmanagerMetrics(r prometheus.Registerer) *RemoteAlertmanager {
 			Name:      "remote_alertmanager_last_state_sync_timestamp_seconds",
 			Help:      "Timestamp of the last successful state sync to the remote Alertmanager in seconds.",
 		}),
+		ConfigSizeBytes: promauto.NewHistogram(prometheus.HistogramOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      "grafana_alerting_alertmanager_config_size_bytes",
+			Help:      "The size of the grafana alertmanager configuration file in bytes",
+			Buckets:   prometheus.ExponentialBuckets(512, 2, 10),
+		}),
```
Comment (santihernandezc) [inline pkg/services/ngalert/metrics/remote_alertmanager.go:90]: This metric looks like a fit for the [Grafana Alertmanager metrics](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/metrics/alertmanager.go), not the remote AM ones.

We can use a gauge to expose the size of the AM config when we apply it, that way we'll know about it even if the `remote.alertmanager.enabled` is set to `false`.

[This method](https://github.com/grafana/grafana/blob/9b2c2be82d643632cbc48112595c068e89851c0f/pkg/services/ngalert/notifier/alertmanager.go#L286-L298) looks interesting, it's called every time we apply a config in the internal AM, we could update the gauge there.

**Review by santihernandezc — CHANGES_REQUESTED** [review]

File: pkg/services/ngalert/metrics/remote_alertmanager.go (hunk 1 above)
Comment (titolins) [reply pkg/services/ngalert/metrics/remote_alertmanager.go]: Good points! Changed now 👍

File: pkg/services/ngalert/metrics/alertmanager.go [hunk 2]
```diff
@@ -53,9 +54,13 @@ func NewAlertmanagerConfigMetrics(r prometheus.Registerer) *AlertmanagerConfigMe
 			Name: "alertmanager_config_object_matchers",
 			Help: "The total number of object_matchers",
 		}),
+		ConfigSizeBytes: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "alertmanager_config_size_bytes",
+			Help: "The size of the grafana alertmanager configuration file in bytes",
```
Comment (santihernandezc) [inline pkg/services/ngalert/metrics/alertmanager.go]:
```suggestion
			Help: "The size of the grafana alertmanager configuration in bytes",
```

**Review by santihernandezc — APPROVED** [review]
Nice! 🎉 
A few comments, once those are addressed, the PR is good to go!

**Pull request merged**